submissions/vovan-zt/TiniWorld2/index.js

```diff
@@ -0,0 +1,84 @@
+class WorldInhabitant {
+	constructor(species, name, gender, saying) {
+		this.species = species;
+		this.name = name;
+		this.gender = gender;
+		this.saying = saying;
+	}
+
+	getMeaning() {
+		return ["species", "name", "gender", "saying"]
+			.map((property) => {
+				if (property === "species") {
+					return this[property].toUpperCase();
+				}
+				return this[property];
+			})
+			.join("; ");
+	}
+}
+class Human extends WorldInhabitant {
+	constructor(name, gender, saying, friends) {
+		super("human", name, gender, saying);
+		this.legs = 2;
+		this.hands = 2;
+		this.friends = friends;
+	}
+
+	getMeaning() {
+		return `${super.getMeaning()};I have ${this.legs} legs and ${
+			this.hands
+		} hands; ${this.friends ? this.friends.name : "no friends"}`;
+	}
+}
+class Man extends Human {
+	constructor(name, saying, friends) {
+		super(name, "man", saying, friends);
+	}
+}
+class Woman extends Human {
+	constructor(name, saying, friends) {
+		super(name, "woman", saying, friends);
+	}
+}
+class CatWoman extends Woman {
+	constructor(name) {
+		super(name);
+		this.species = "CatWoman";
+		this.saying = cat.saying;
+	}
+}
+class Animal extends WorldInhabitant {
+	constructor(species, name, gender, saying) {
+		super(species, name, gender, saying);
+		this.paws = 4;
+	}
+
+	getMeaning() {
+		return `${super.getMeaning()}; animal has ${this.paws} paws`;
```

Comment:
Here breaking template literal string may not make much sense.

Reply:
Can I do it like this?

	getMeaning() {
		return super.getMeaning() + `<strong> animal has ${this.paws} paws <strong>`;
	}

Reply:
String is short. There are no any critical repetitions indeed.
Could be.
I do not like mixing string literals and string concatenations with `+`, but it actually depends on particular situation, so not a hard requirement. Just take care of your code readers. One day you will have to read someone else's code.

Reply:
getMeaning() {
		return (
			[super.getMeaning(),
      `<strong> animal has ${this.paws} paws <strong>`]
		);
	}

Reply:
I didn't even think it was bad practice. Need to remember.

Reply:
Not necessarily bad. It is all about readability (by others) and maintainability.
Putting string components into an array is also not a panacea. On big or frequent operations it may cause performance issues. From the other hand on big or frequent operations string literals will have their own issues.
In this exercise we just didn't need to optimize for performance, so we care about readability. But as project grows things may change.